### PR TITLE
[WIP] Refactored cloud tests to use 'assertCreateInstance'

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Refactor the cloud tests to use `assertCreateInstance`
Combine all digital ocean tests into the same directory
Separated from https://github.com/saltstack/salt/pull/54972/

This refactor will cause `test_instance` to fail if salt isn't bootstrapped correctly after deploying an image -- We want that :+1:

### CODE REVIEW

**Comment by Akm0d** [comment]
The cloud test suite on `master` currently shows a bunch of false positives.  
With this refactor we properly blow up on bootstrap failures and creation errors:
https://jenkinsci.saltstack.com/view/Akm0d%20Jobs/job/Develop%20Py3%20Cloud/280/

**Comment by Akm0d** [comment]
https://github.com/saltstack/salt/pull/55046 is an attempt to solve the win2012r2 psexec and winrm tests

File: tests/integration/cloud/clouds/test_vultrpy.py [hunk 1]
```diff
@@ -47,55 +67,52 @@ def test_list_sizes(self):
         size_list = self.run_cloud('--list-sizes {0}'.format(self.PROVIDER))
         self.assertIn('2048 MB RAM,64 GB SSD,2.00 TB BW', [i.strip() for i in size_list])
 
-    # Commented for now, Vultr driver does not yet support key management
```
Comment (dwoz) [inline tests/integration/cloud/clouds/test_vultrpy.py]: This can be removed.

**Review by dwoz — CHANGES_REQUESTED** [review]

File: tests/integration/cloud/clouds/test_vultrpy.py (hunk 1 above)
Comment (Akm0d) [reply tests/integration/cloud/clouds/test_vultrpy.py]: It is :+1:

**Comment by Akm0d** [comment]
integration.cloud.clouds.test_msazure.AzureTest.test_instance
```Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/cloud/clouds/test_msazure.py", line 60, in test_instance
    self.assertCreateInstance(timeout=TIMEOUT)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 79, in assertCreateInstance
    self.assertInstanceExists(creation_ret=stdout, instance_name=instance_name, stderr=stderr)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 93, in assertInstanceExists
    self.fail('\n'.join(stderr))
AssertionError: [WARNING ] There's no master defined on the 'cloud-test-azu-hvbbe' VM settings.
No ECDSA host key is known for 40.112.223.183 and you have requested strict checking.
Host key verification failed.```

**Comment by Akm0d** [comment]
integration.cloud.clouds.test_gogrid.GoGridTest.test_instance 
```Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/cloud/clouds/test_gogrid.py", line 24, in test_instance
    self.assertCreateInstance(timeout=TIMEOUT)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 79, in assertCreateInstance
    self.assertInstanceExists(creation_ret=stdout, instance_name=instance_name, stderr=stderr)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 93, in assertInstanceExists
    self.fail('\n'.join(stderr))
AssertionError: [WARNING ] SSL Error on 13 ('50.18.48.147', 443): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:877)
[WARNING ] There's no master defined on the 'cloud-test-gog-vedul' VM settings.
[WARNING ] SSL Error on 9 ('184.169.137.160', 443): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:877)
[ERROR   ] There was a profile error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:877)```

**Comment by Akm0d** [comment]
integration.cloud.clouds.test_vultrpy.VultrTest.test_instance 
```Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/cloud/clouds/test_vultrpy.py", line 115, in test_instance
    self.assertCreateInstance(timeout=TIMEOUT + 300)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 79, in assertCreateInstance
    self.assertInstanceExists(creation_ret=stdout, instance_name=instance_name, stderr=stderr)
  File "/tmp/kitchen/testing/tests/integration/cloud/helpers/cloud_test_base.py", line 93, in assertInstanceExists
    self.fail('\n'.join(stderr))
AssertionError: [WARNING ] There's no master defined on the 'cloud-test-vul-jgfqa' VM settings.
Warning: Permanently added '45.63.6.140' (ECDSA) to the list of known hosts.
Permission denied, please try again.
Permission denied, please try again.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).```

**Review by Ch3LL — CHANGES_REQUESTED** [review]
looks like this has some merge conflicts that need to be resolved and pre-commit failing

**Comment by Ch3LL** [comment]
Is this still being worked on? If not can this be closed?

**Comment by Ch3LL** [comment]
Closing due to inactivity

**Pull request closed without merging**